disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -98,3 +103,19 @@ listOfSizedWithIndex :: Int -> Int -> (Int -> Gen a) -> Gen [a]
 listOfSizedWithIndex from to g =
   chooseSize from to >>=
     mapM g . enumFromTo 0
+
+newtype GenSeed =
+  GenSeed {
+    unGenSeed :: Int
+  } deriving (Eq, Show)
+
+-- | Deterministic generator with a default size of 100.
+genDeterministic :: GenSeed -> Gen a -> IO a
+genDeterministic = genDeterministic' 100
+
+-- | Deterministic generator, always produces the same output for the same
+-- seed.
+genDeterministic' :: Int -> GenSeed -> Gen a -> IO a
+genDeterministic' size (GenSeed seed) (MkGen g) =
+  let r = mkQCGen seed in
+  pure $ g r size
```

Comment:
I might be misreading this, but it doesn't look like this function needs to be `IO` given you're manually using `pure`?